app/code/Magento/Eav/Model/Config.php

```diff
@@ -482,19 +492,19 @@ protected function _initAttributes($entityType)
             $entityType
         )->getData();
 
-        $this->_attributeData[$entityTypeCode] = [];
+        $this->_attributeData[$storeId][$entityTypeCode] = [];
         foreach ($attributes as $attribute) {
             if (empty($attribute['attribute_model'])) {
                 $attribute['attribute_model'] = $entityType->getAttributeModel();
             }
             $attributeObject = $this->_createAttribute($entityType, $attribute);
             $this->saveAttribute($attributeObject, $entityTypeCode, $attributeObject->getAttributeCode());
-            $this->_attributeData[$entityTypeCode][$attribute['attribute_code']] = $attributeObject->toArray();
+            $this->_attributeData[$storeId][$entityTypeCode][$attribute['attribute_code']] = $attributeObject->toArray();
         }
         if ($this->isCacheEnabled()) {
             $this->_cache->save(
-                $this->serializer->serialize($this->_attributeData[$entityTypeCode]),
-                self::ATTRIBUTES_CACHE_ID . $entityTypeCode,
+                $this->serializer->serialize($this->_attributeData[$storeId][$entityTypeCode]),
+                self::ATTRIBUTES_CACHE_ID . '-' . $this->getStoreId() . '-' . $entityTypeCode,
```

Comment:
I think we should use $storeId variable instead
```suggestion
                self::ATTRIBUTES_CACHE_ID . '-' . $storeId . '-' . $entityTypeCode,
```

Reply:
yep, and in initAttributesFromCache method too